include/nbl/builtin/hlsl/algorithm.hlsl

```diff
@@ -11,6 +11,84 @@ namespace nbl
 namespace hlsl
 {
 
+namespace impl
+{
+#ifdef __HLSL_VERSION
+
+    // TODO: use structs
+
+    template<typename T>
+    NBL_CONSTEXPR_INLINE_FUNC void swap(NBL_REF_ARG(T) lhs, NBL_REF_ARG(T) rhs)
```

Comment:
could be done with two overloads of `enable_if_t<is_fundamental_v<T>,void>` and `enable_if_t<!is_fundametal_v<T>,void>` if the compiler lets you

P.S. such bitwise swaps can also be done on floats, if you bitcast them into size matching uints first